web/src/reminders_overlay_ui.ts

```diff
@@ -0,0 +1,130 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+
+import render_reminder from "../templates/reminder.hbs";
+import render_reminders_overlay from "../templates/reminders_overlay.hbs";
+
+import * as browser_history from "./browser_history.ts";
+import * as message_reminder from "./message_reminder.ts";
+import type {Reminder} from "./message_reminder.ts";
+import * as messages_overlay_ui from "./messages_overlay_ui.ts";
+import * as overlays from "./overlays.ts";
+import * as timerender from "./timerender.ts";
+
+type ReminderRenderContext = Reminder & {
+    formatted_send_at_time: string;
+};
+
+export const keyboard_handling_context = {
+    get_items_ids() {
+        const reminders_ids = [];
+        const sorted_messages = sort_reminders(message_reminder.reminders_data);
```

Comment:
Rather than having `message_reminder` export its internal data structure, we should just have a `message_reminder.get_all_reminders()` function that returns them in mutable form. That avoids bugs where logic here for mutating things messes stuff up.

Also `sorted_reminders` would be a much better local variable name; "messages" is not a keyword we want to be using for things that are not `Messages`. (I guess that's an existing readability bug in the scheduled messages code).

Reply:
Done